### DISCUSSION TITLE
Using DeepONet for Boundary Condition Mapping and Custom Function Spaces

### DISCUSSION THREAD

**Discussion by skhaleghir** [lululxvi/deepxde#1886] [Q&A]
Hello, I'd like to first thank you Professor Lu for creating and sharing this remarkable tool with the community.

I am currently working on training a neural operator to map the flux at a boundary condition (as a function of time) to the concentration profile in a time-dependent 1D diffusion problem (details attached). While implementing this, I encountered the following error:
"flux_function() missing 1 required positional argument: 'v'"

I would greatly appreciate your guidance on resolving this issue.

Additionally, I have a second question: Is there documentation or guidance on defining custom function spaces? I aim to train the model under specific flux conditions that cannot be represented using the existing options, such as "PowerSeries" or GRF.

Thank you for your time and assistance!

Saeed

<img width="708" alt="1D_diffusion" src="https://github.com/user-attachments/assets/7ad5f70f-4fad-43f7-b97b-0fe0716e2a76">

**Comment by lululxvi** [comment]
RobinBC doesn't support input function not. I guess we just need to change this line
https://github.com/lululxvi/deepxde/blob/3544fdfd4e43ffb920d2b191a53d66946226199e/deepxde/icbc/boundary_conditions.py#L105
to
```
X[beg:end], outputs[beg:end], aux_var[beg:end]
```

You need to implement a new class to generate functions. Check the code of PowerSeries or GRF, and it should be easy to impelement.

**Comment by skhaleghir** [comment]
Thank you, Lu, for the suggestions. I will try them and let you know if I have any further questions.
